src/Formula/Parsing/Delayed.hs

```diff
@@ -14,17 +14,23 @@ import LogicTasks.Helpers (reject)
 import Data.Typeable (Typeable)
 import GHC.Generics (Generic)
 
-newtype Delayed a = Delayed {unDelayed :: String} deriving (Eq, Typeable, Generic)
+newtype Delayed a = Delayed String deriving (Eq, Typeable, Generic)
+
+instance Show (Delayed a) where
+  show (Delayed str) = str
```

Comment:
Is there even a need for a `Show` instance for `Delayed` now?

Reply:
At the moment it is used for the `Hashable (Delayed a)` instance in Autotool. At the "expense" of a `hashable` dependency the instance could be defined here instead (making it also non-orphan).

Reply:
But why is `Hashable (Delayed a)` even needed? I guess I am still struggling to get to grips with the role of `Delayed`. There shouldn't be a task type that has `Delayed` as its input type, nor inside a configuration or instance type, right? That is the whole point of having things like
```haskell
type CnfString = String
instance Partial TruthTableMaxterm MaxInst CnfString where
```
in Autotool, not
```haskell
instance Partial TruthTableMaxterm MaxInst (Delayed Cnf) where
```
right?

Reply:
You are right. It is not needed. I transformed the actual string input too early in a few cases where `completeGrade` is used inside of `toReporterWithGlobalPics` which requires `Hashable` inputs.

Reply:
Does something similar apply to the `Eq, Typeable, Generic` instances?

Due to the peculiar role of `Delayed`, the pull request's intention really was to "isolate" it somewhat from the rest of the code via making it as abstract as possible. Any provided type class instance of course makes it a bit less abstract, since one can then learn something about internals of a delayed value.